lib/node_modules/@stdlib/repl/lib/parse_key.js

```diff
@@ -0,0 +1,267 @@
+
+
+// VARIABLES //
+
+// Control sequence prefix in ALT combinations:
+var ALT_CSI = '\u001B';
+
+// Properties in parsed keybinding:
+var KEYBINDING_PROPS = [ 'name', 'ctrl', 'shift', 'meta' ];
+
+/**
+* Asserts if a sequence is an unrecognized symbol.
+*
+* ## Notes
+*
+* -   `SHIFT+<symbol>` will result in a keybinding with the name of the resulting symbol and not the original symbol with shift enabled.
+*
+* @private
+* @name isUnrecognizedSymbol
+* @type {Function}
+*/
+var isUnrecognizedSymbol = containsFactory([
+	'`',
+	'~',
+	'!',
+	'@',
+	'#',
+	'$',
+	'%',
+	'^',
+	'&',
+	'*',
+	'(',
+	')',
+	'-',
+	'_',
+	'=',
+	'+',
+	'{',
+	'}',
+	'[',
+	']',
+	'|',
+	'\\',
+	':',
+	';',
+	'"',
+	'\'',
+	'<',
+	',',
+	'>',
+	'.',
+	'?',
+	'/'
+]);
+
+/**
+* Table mapping unrecognized control sequences to their possible key objects.
+*
+* ## Notes
+*
+* -   The following map is recorded as observed in gnome.
+*
+* @private
+* @name COMBINATIONAL_KEYMAP
+* @type {Object}
+*/
+var CTRL_SEQUENCES_KEYMAP = {
+	'\u001F': [
+		{
+			'name': '/',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': '-',
+			'ctrl': true,
+			'shift': false,
+			'meta': true
+		},
+		{
+			'name': '7',
+			'ctrl': true,
+			'shift': false,
+			'meta': true
+		}
+	],
+	'\u001C': [
+		{
+			'name': '\\',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': '\\',
+			'ctrl': true,
+			'shift': true,
+			'meta': false
+		},
+		{
+			'name': '|',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': '4',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		}
+	],
+	'\u001D': [
+		{
+			'name': ']',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': ']',
+			'ctrl': true,
+			'shift': true,
+			'meta': false
+		},
+		{
+			'name': '}',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': '5',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		}
+	],
+	'\u001E': [
+		{
+			'name': '6',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		},
+		{
+			'name': '6',
+			'ctrl': true,
+			'shift': true,
+			'meta': false
+		},
+		{
+			'name': '^',
+			'ctrl': true,
+			'shift': false,
+			'meta': false
+		}
+	]
+};
+
+
+// FUNCTIONS //
+
+/**
+* Parses unrecognized control sequences emitted by `ALT+<symbol>` combinations.
+*
+* @private
+* @param {string} seq - control sequence
+* @returns {(Object|boolean)} key object or false if not a meta symbol sequence
+*/
+function parseMetaSymbolSequence( seq ) {
+	if (
+		!seq ||
+		seq.length !== 2 ||
+		seq[ 0 ] !== ALT_CSI ||
+		isUnrecognizedSymbol( seq[ 1 ] )
+	) {
+		return false;
+	}
+	return [
+		{
+			'name': seq[ 1 ],
+			'ctrl': false,
+			'shift': false,
+			'meta': true
+		}
+	];
+}
+
+
+// MAIN //
+
+/**
+* Parses a readline keypress object.
+*
+* @param {string} key - readline keypress object
+* @returns {(Array<Object>|boolean)} list of possible key objects
```

Comment:
These types seem off. Is `key` an object or a string? And it is not obvious that the function can return a `boolean`.

Reply:
Okay. `parseMetaSymbolSequence` can return a boolean. Would it make more sense to make the types consistent? For example, return an empty array? Alternatively, rather than a boolean, it is more common for us to return `null`.

Reply:
I forgot to update the jsdoc, sorry for the confusion. Yes, the main parse_key function doesn't return a boolean. 

I updated `parseMetaSymbolSequence` to return `null` instead. If it returns `null` means it's not a meta symbol sequence. So we just return the original key object.

Should `parseMetaSymbolSequence` return an empty array? The only problem would be, we won't be able to check directly if it returned a sequence or not (using `if ( out )`) but would have to check if the array is empty. `null` should be fine(?).